scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/Worker.java

```diff
@@ -100,13 +129,25 @@ private TaskActionsQueue queueFirstActionForEachTask(Map<TaskId, SortedSet<Strea
      * At each iteration, runs a single action from |actions| queue if any
      * available.
      */
-    private void performActionsUntilStopRequested(TaskActionsQueue actions) {
-        while (!workerConfiguration.transport.shouldStop()) {
-            try {
-                actions.runNextAction();
-            } catch (InterruptedException e) {
-                // Ignore InterruptedException
-            }
+    @SuppressWarnings("deprecation")
+    private void performActionsUntilStopRequested(Collection<TaskAction> actions) {
+        if (shouldStop()) {
+            return;
+        }
+
+        ScheduledExecutorService executorService = getExecutorService();
+        try {
+            executorService.invokeAll(actions.stream().map(a -> makeCallable(a)).collect(Collectors.toSet()));
+            do {
+                // pretty short poll, to allow for reasonably fast switchover
+                // iff using "polled" shutdown (WorkerTransport::shouldStop).
+                executorService.awaitTermination(50, MILLISECONDS);
+                if (workerConfiguration.transport.shouldStop()) {
+                    stop();
```

Comment:
I think you can add break to this if, no?

Reply:
No. We still need to loop until the executor says is it done (terminated).

Reply:
I see.